### DISCUSSION TITLE
How to extract subtitles around specific YT video timepoint?

### DISCUSSION THREAD

**Discussion by lofti198** [fent/node-ytdl-core#1088] [Q&A]
Hi folks! Can I get subtitles around specific YT video timepoint? Do I need to load all subtitles for this puprose?

**Comment by TimeForANinja** [comment]
all subtitles are a single file (at least per language)
so you'd have to fetch all (but since it's only text we're taking kilobytes at max)

last i checked the link to them was somewhere in the info.json file
they should be in some kinda srt or xml format

**Reply by lofti198** [reply]
I use the following code:

`const ytdl = require('ytdl-core');
const axios = require('axios');
const { XMLParser, XMLBuilder, XMLValidator} = require('fast-xml-parser');


const getSubTitles = async (url) =>
{
  try {
    const parser = new XMLParser();
    
    const info = await ytdl.getInfo(url);
    // console.log( info)
    if (!info.player_response.captions) return null;
    const tracks = info.player_response.captions.playerCaptionsTracklistRenderer.captionTracks;

    if (!tracks || !tracks.length) return null;

    const parsedTracks = await Promise.all(tracks.map(async (track) => (
        {
            lang: track.languageCode,
            content: parser.parse((await axios.get(track.baseUrl)).data, {
            ignoreAttributes: false,
        }),
    })));
    
    // console.log(parsedTracks[0].transcript)

    return parsedTracks[0].content.transcript.text
        // .map((elem) => (       
        //     {        
        // Text: elem['#text'],
        // TimePoint: Math.floor(elem['@_start']),
        // }));
  }
  catch(e) {
    console.log("Exception happened:\n" );
    console.log(e)
    return null;
  }
}

//console.log( getSubTitles("https://www.youtube.com/watch?v=ONG26-2mIHU"));

getSubTitles("https://www.youtube.com/watch?v=ONG26-2mIHU").then((value) => {
    console.log(value);
    // expected output: "Success!"
  });
  `

But fetching just text without timestamps (for each particular subtitle line). Any workarounds how to fetch lines with timepoints?

![image](https://user-images.githubusercontent.com/42960488/166133486-b6e37865-0334-4ede-b218-8aa0904406c1.png)
